src/lib/frontend/endpoints/Endpoints.ts

```diff
@@ -10,27 +8,29 @@ import type {OrderDetailsPresentation} from '$lib/frontend/presentations/OrderDe
 import {API_BASE_ENDPOINT, BASE_HEADERS} from '$lib/frontend/Constants'
 import type {BusinessContactRequest} from '$lib/frontend/requests/BusinessContactRequest'
 import type {GenericSuccessResponsePresentation} from '$lib/frontend/presentations/GenericSuccessResponsePresentation'
+import type {CheckoutRequest, CheckoutResponse} from '$lib/frontend/requests/CheckoutRequest'
+import {EndpointError} from '$lib/frontend/endpoints/EndpointError'
 
-export async function createCheckoutSession(parameters: CheckoutRequest): Promise<CheckoutResponse | undefined> {
+export async function createCheckoutSession(parameters: CheckoutRequest): Promise<CheckoutResponse> {
```

Comment:
I am curious, why an error instead of undefined?

Reply:
I think you are right, the undefined approach is better and avoid the hassle of dealing with the try catch